### ISSUE TITLE
BlockPolicyNet that works with Control variables with upper and lower bounds

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#66]
The `BlockPolicyNet` currently assumes that a Control can have any (unbounded) real value.

https://github.com/scikit-agent/scikit-agent/blob/main/src/skagent/ann.py#L33

The Control object can specify upper and lower bounds, which can have a numerical value, but also can be state or parameter dependent. (I.e. they can be 'expressions' of other variables in scope).

While it is possible to find clever workarounds for this (i.e., having a real valued control variable, then adding structural equations that wrap the control in a sigmoid or rectilinear function, then transform that into a value that is within the bounds), it's probably better if we support the more expressive usage of the Control object and have the BlockPolicyNet do the structuring for us.

Here are examples of simple examples that demonstrate the issue. There may well be better examples in the BENCHMARKs.

Double bounded:
```
DBlock(
    **{
        "name": "upper bound",
        "shocks": {
            "theta": (Normal, {"mu": 0, "sigma": 1}),
        },
        "dynamics": {
            "m": lambda m, theta: a + theta,
            "c": Control(["m"], upper_bound=lambda m: (m ** 2) ** .4 , lower_bound=lambda m: - (m ** 2) ** .4, agent="tester"),
            "u": lambda c: -(m + c)**2,
        },
        "reward": {"u": "tester"},
    }
)
```
Both bounds should bind, whether `m` is positive or negative.

Upper bounded only:
```
DBlock(
    **{
        "name": "upper bound",
        "shocks": {
            "theta": (Normal, {"mu": 0, "sigma": 1}),
        },
        "dynamics": {
            "m": lambda m, theta: a + theta,
            "c": Control(["m"], upper_bound=lambda m: m, agent="tester"),
            "u": lambda c: c,
        },
        "reward": {"u": "tester"},
    }
)
```
Upper bound binds.

Lower bounded only:
```
DBlock(
    **{
        "name": "upper bound",
        "shocks": {
            "theta": (Normal, {"mu" : 0, "sigma": 1}),
        },
        "dynamics": {
            "m": lambda m, theta:ma + theta,
            "a": Control(["m"], lower_bound=0, agent="tester"),
            "u": lambda a, m: m - a,
        },
        "reward": {"u": "tester"},
    }
)
```
Lower bound binds.

**Comment by sbenthall** [comment]
Softplus is the activation function to use for single-bounded controls:
https://docs.pytorch.org/docs/stable/generated/torch.nn.Softplus.html#torch.nn.Softplus
